src/contexts/video-action-context/video-action-reducer.jsx

```diff
@@ -0,0 +1,69 @@
+const videoActionReducer = (state, action) => {
+  switch (action.type) {
+    case "GET_LIKED_VIDEOS":
+      return {
+        ...state,
+        liked: action.payload,
+        likedSize: action.payload.length,
+      };
+    case "ADD_TO_LIKED":
+      return {
+        ...state,
+        liked: action.payload,
+        likedSize: action.payload.length,
+      };
+    case "REMOVE_FROM_LIKED":
```

Comment:
Instead of creating three cases for "GET_LIKED_VIDEOS", "ADD_TO_LIKED", "REMOVE_FROM_LIKED", you can simply create and "UPDATE_LIKED" case that updates the liked state because all the 3 do the same thing. They simply take the payload and add to liked.

The same goes for watch later, history etc.. this way your reducer cases will be very less

Reply:
Oh right, thanks Piyush!!  i was thinking on optimizing this. :)

Reply:
Made the changes!